scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java

```diff
@@ -98,23 +99,25 @@
 
     private SpaceInstanceInfo cacheSpaceInstanceInfo;
 
-    private transient StringBuffer clientLogs = new StringBuffer();
-
     private transient ExecutorService executorTransfer = Executors.newFixedThreadPool(getFileTransferThreadPoolSize(),
                                                                                       new NamedThreadFactory("FileTransferThreadPool"));
 
+    private transient TaskLogger taskLogger;
+
     /**
      * Mainly for testing purposes
      */
     TaskProActiveDataspaces() {
 
     }
 
-    public TaskProActiveDataspaces(TaskId taskId, NamingService namingService, boolean isRunAsUser) throws Exception {
+    public TaskProActiveDataspaces(TaskId taskId, NamingService namingService, boolean isRunAsUser,
+            TaskLogger taskLogger) throws Exception {
         this.taskId = taskId;
         this.namingService = namingService;
         this.runAsUser = isRunAsUser;
         this.linuxOS = OperatingSystem.getOperatingSystem() == OperatingSystem.unix;
+        this.taskLogger = taskLogger;
```

Comment:
The taskLogger attribute could have manipulated in a smoother way if it was implementing Optionnal<>

Reply:
I am not a fan of Optional (to say the least)